.gitlab-ci.yml

```diff
@@ -96,6 +97,18 @@ default:
     - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
     - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
 
+.test-refs-no-trigger-no-master:   &test-refs-no-trigger-no-master
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "pipeline"
+      when: never
+    - if: $CI_COMMIT_REF_NAME == "master"
+      when: never
+    - if: $CI_PIPELINE_SOURCE == "web"
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+    - if: $CI_COMMIT_REF_NAME == "tags"
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
```

Comment:
_philosohically:_ since this is a perf guard and it should prevent the slow code getting merged, does it make sense to check it again on tags?

Reply:
Resolved [here](https://github.com/paritytech/substrate/pull/8519/commits/7ed876a3e2dd43398832311b1b80f8aa3a3d8f89).